docs/tpm2-pkcs11_object_auth_model.md

```diff
@@ -0,0 +1,528 @@
+a policy. These are user created keys and objects that are used in user
+operations like encryption, decryption, signing, verification, etc. These are
+secondary transient objects.
+
+## NV object diagram
+Overview of the use of the TPM2 NV indexes in the implementation.
+<p>
+  <img src="illustrations/object_diagram_nv_objects.png"
+</p>
+<br>
+<br>
+The illustration above is the key/object diagram showing the relationship of the
+tpm2-pkcs11 objects. The SO-PIN and USER-PIN objects are NV index objects while
+the Wrapping-Key-Sealing object and the USER(keys) objects are secondary
+transient objects.
+
+## NV policy diagram
+Overview of policies used to satisfy standards requirements with NV objects used
+as SO/ USER PIN reference objects.
+<p>
+  <img src="illustrations/policy_model_nv_objects.png"
+</p>
+<br>
+<br>
+The illustration above is the policy diagram showing the relationship of the
+authorization model used by the tpm2-pkcs11 objects where the SO-PIN and the
+USER-PIN objects are NV indexes and the Wrapping-Key-Sealing object and the USER
+(keys) objects are transient secondary objects.
+
+## Transient object diagram
+Overview of the use of TPM2 transient objects in the implementation.
+<p>
+  <img src="illustrations/object_diagram_transient_objects.png"
+</p>
+<br>
+<br>
+The illustration above is the key/object diagram showing the relationship of the
+tpm2-pkcs11 objects. The SO-PIN and USER-PIN objects along with the
+Wrapping-Key-Sealing object and the USER(keys) objects are secondary transient
+objects.
+
+## Transient policy diagram
+Overview of the policies used to satisfy standards requirements with TPM2
+transient object used as SO/ USER PIN reference objects.
+<p>
+  <img src="illustrations/policy_model_transient_objects.png"
+</p>
+<br>
+<br>
+The illustration above is the policy diagram showing the relationship of the
+authorization model used by the tpm2-pkcs11 objects where the SO-PIN and the
+USER-PIN objects along with the Wrapping-Key-Sealing object and the USER (keys)
+objects are transient secondary objects.
+
+## Important note on authorization
+While both authorization models discussed resolve the issue of isolation of SO
+and USER roles there is a serious problem with the authorization model where the
+SO-PIN and USER-PIN reference objects are transient secondary objects. In that,
+when the authorization of the objects is changed it creates a new sensitive
+portion of the object while the old sensitive information cannot be invalidated.
+To that effect, anyone who has a cached copy of an older object whose pin may
+have been compromised can still access the USER objects.
+<br>
+<br>
+The issue can be resolved by using NV indices as SO-PIN and USER-PIN reference
+objects instead. Although, we still need the sealing object to seal the
+wrapping secret for bringing the existing TPM objects into the pkcs11 fold as
+well as the USER(keys) objects. However their auth will be tied to USER-PIN NV
```

Comment:
> Although, we still need the sealing object to seal the
> wrapping secret for bringing the existing TPM objects into the pkcs11 fold as
> well as the USER(keys) objects.

What does this sentence mean? From context, I think maybe it's trying to say something like "The User (Keys) and Wrappeing-Key-Sealing objects will still be transient objects"?

Reply:
Correct.

Reply:
The ultimate goal is to burn 2 nv indices, but in the off chance we can't we'll have the fallback mode. Or for people where burning 2 nv indices is not tolerated.

Reply:
or for folks who would want to use the NULL hierarchy

Reply:
or even endorsement hierarchy